### PR TITLE
Add span.kind attribute to newrelic exporter

### PR DESCRIPTION
**Pull request by jamal**
**Description:** Adds `span.kind` attribute when `Kind` is set in the Span.

**Testing:** Added tests to confirm the change. Was also able to test against otelcolcontrib locally and confirmed that the span is being published with this new attribute.

### CODE REVIEW

File: exporter/newrelicexporter/transformer.go [hunk 1]
```diff
@@ -111,6 +112,11 @@ func (t *transformer) SpanAttributes(span *tracepb.Span) map[string]interface{}
 		attrs["error"] = true
 	}
 
+	// Add span kind if it is set
+	if span.Kind > tracepb.Span_SPAN_KIND_UNSPECIFIED {
```
Comment (bogdandrutu) [inline exporter/newrelicexporter/transformer.go:116]: !=?

**Review by bogdandrutu — COMMENTED** [review]
@MrAlias can you review this PR?

File: exporter/newrelicexporter/transformer.go (hunk 1 above)
Comment (MrAlias) [reply exporter/newrelicexporter/transformer.go:116]: Agreed. There is no guarantee on the underlying ordering of the checked value. We should just check for inequality here.

```suggestion
	if span.Kind != tracepb.Span_SPAN_KIND_UNSPECIFIED {
```

**Review by MrAlias — APPROVED** [review]
Only issue is the minor one with the comparison, otherwise looks good :+1:

**Comment by jamal** [comment]
Makes sense, updated it to use `!=`. Thanks!

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**